tools/nimgrep.nim

```diff
@@ -569,45 +1108,69 @@ for kind, key, val in getopt():
+    of "ignorestyle", "ignore-style", "y": incl(options, optIgnoreStyle)
+    of "nworkers", "n":
+      if val == "":
+        nWorkers = countProcessors()
+      else:
+        nWorkers = parseNonNegative(val, key)
+    of "ext": walkOpt.extensions.add val.split('|')
+    of "noext", "no-ext": walkOpt.skipExtensions.add val.split('|')
+    of "excludedir", "exclude-dir",   "ed": walkOpt.excludeDir.add val
+    of "includedir", "include-dir",   "id": walkOpt.includeDir.add val
+    of "includefile", "include-file", "if": walkOpt.includeFile.add val
+    of "excludefile", "exclude-file", "ef": walkOpt.excludeFile.add val
+    of "match": searchOpt.checkMatch = val
+    of "nomatch", "notmatch", "not-match", "no-match":
+      searchOpt.checkNoMatch = val
+    of "bin":
+      case val
+      of "no": searchOpt.checkBin = biNo
+      of "yes": searchOpt.checkBin = biYes
+      of "only": searchOpt.checkBin = biOnly
+      else: reportError("unknown value for --bin")
+    of "text", "t": searchOpt.checkBin = biNo
+    of "count": incl(options, optCount)
+    of "sorttime", "sort-time", "s":
+      sortTime = true
+      case normalize(val)
+      of "": discard
+      of "asc", "ascending": sortTimeOrder = SortOrder.Ascending
+      of "desc", "descending": sortTimeOrder = SortOrder.Descending
+      else: reportError("invalid value '" & val & "' for --sortTime")
+    of "nocolor", "no-color": useWriteStyled = false
     of "color":
       case val
       of "auto": discard
       of "never", "false": useWriteStyled = false
       of "", "always", "true": useWriteStyled = true
       else: reportError("invalid value '" & val & "' for --color")
-    of "colortheme":
+    of "colortheme", "color-theme":
       colortheme = normalize(val)
       if colortheme notin ["simple", "bnw", "ack", "gnu"]:
         reportError("unknown colortheme '" & val & "'")
     of "beforecontext", "before-context", "b":
-      try:
-        linesBefore = parseInt(val)
-      except ValueError:
-        reportError("option " & key & " requires an integer but '" &
-                    val & "' was given")
+      linesBefore = parseNonNegative(val, key)
     of "aftercontext", "after-context", "a":
-      try:
-        linesAfter = parseInt(val)
-      except ValueError:
-        reportError("option " & key & " requires an integer but '" &
-                    val & "' was given")
+      linesAfter = parseNonNegative(val, key)
     of "context", "c":
-      try:
-        linesContext = parseInt(val)
-      except ValueError:
-        reportError("option --context requires an integer but '" &
-                    val & "' was given")
+      linesContext = parseNonNegative(val, key)
     of "newline", "l": newLine = true
     of "oneline": oneline = true
     of "group", "g": oneline = false
+    of "limit", "m":
+      incl(options, optLimitChars)
+      if val == "":
+        limitChar = 80
+      else:
+        limitChar = parseNonNegative(val, key)
+    of "fit":
```

Comment:
`--fit` doesn't work reliable, eg go to root of nim repo, type:
```
`$ nimgrep -r -t --fit PSym`

![image](https://user-images.githubusercontent.com/2194784/96804535-1ca49d00-13c4-11eb-8f35-6d698b9f3b31.png)

that said, it "almost works" so if it's hard to fix in this PR, just file an issue so it can be fixed later

Reply:
yes, limiting was not implemented for matches itself, only for surrounding context. I'm working on it...

> `wcwidth`

If you mean unicode support, then currently it is not implemented, but I'm trying to do it by using nim's `unicode` module. AFAIK `wcwidth` is for wide 16-bit/32-bit characters that are not recommended even on Windows now, so currently the reasonable option for all the platforms is UTF-8, which exactly what `unicode` does. I think `wcwidth` can be regarded as de-facto deprecated (may be it'll revive in future if fashion changes).

Reply:
see https://github.com/nim-lang/RFCs/issues/279

Reply:
yes, I've finally understood what you mean.
However, fully supporting unicode is a bit too difficult then, I gave up the idea of that at the moment.
Adding unicode will be left for future.

In the current version cropping Ascii matches is supported well.